classes/services/queryBuilders/PKPUserQueryBuilder.inc.php

```diff
@@ -441,267 +443,227 @@ public function getIds() {
+				$words = array_map(function (string $word): string {
+					return '%' . addcslashes($word, '%_') . '%';
+				}, PKPString::regexp_split('/\s+/', $searchPhrase));
 
-		// search phrase
-		if (!empty($this->searchPhrase)) {
-			$words = explode(' ', $this->searchPhrase);
-			if (count($words)) {
-				$q->leftJoin('user_settings as us', 'u.user_id', '=', 'us.user_id');
-				$q->leftJoin('user_interests as ui', 'u.user_id', '=', 'ui.user_id');
-				$q->leftJoin('controlled_vocab_entry_settings as cves', 'ui.controlled_vocab_entry_id', '=', 'cves.controlled_vocab_entry_id');
 				foreach ($words as $word) {
-					$word = strtolower(addcslashes($word, '%_'));
-					$q->where(function($q) use ($word) {
-						$q->where(Capsule::raw('lower(u.username)'), 'LIKE', "%{$word}%")
-							->orWhere(Capsule::raw('lower(u.email)'), 'LIKE', "%{$word}%")
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', IDENTITY_SETTING_GIVENNAME);
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
-							})
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', IDENTITY_SETTING_FAMILYNAME);
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
+					$query->where(function(Builder $query) use ($word) {
+						$query->whereRaw('LOWER(u.username) LIKE LOWER(?)', [$word])
+							->orWhereRaw('LOWER(u.email) LIKE LOWER(?)', [$word])
+							->orWhereExists(function (Builder $query) use ($word): void {
+								$query->from('user_settings', 'us')
+									->whereColumn('us.user_id', '=', 'u.user_id')
+									->whereIn('us.setting_name', [
+										IDENTITY_SETTING_GIVENNAME,
+										IDENTITY_SETTING_FAMILYNAME,
+										'preferredPublicName',
+										'affiliation',
+										'biography',
+										'orcid'
+									])
+									->whereRaw('LOWER(us.setting_value) LIKE LOWER(?)', [$word]);
 							})
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', 'preferredPublicName');
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
-							})
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', 'affiliation');
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
-							})
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', 'biography');
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
-							})
-							->orWhere(function($q) use ($word) {
-								$q->where('us.setting_name', 'orcid');
-								$q->where(Capsule::raw('lower(us.setting_value)'), 'LIKE', "%{$word}%");
-							})
-							->orWhere(Capsule::raw('lower(cves.setting_value)'), 'LIKE', "%{$word}%");
+							->orWhereExists(function (Builder $query) use ($word): void {
+								$query->from('user_interests', 'ui')
```

Comment:
The `/main` branch has removed search over this table. Should I remove here as well? I just left to avoid changing things too much.

Reply:
Actually that was an oversight in `main` -- could you re-add it there too?